In [1]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Q1: Jumlah baris dan kolom dataset Titanic serta nama kolom

In [4]:
df.shape
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

Dataset Titanic memiliki 891 baris dan 12 kolom.
Kolom yang tersedia adalah PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, dan Embarked.

Q2: Kolom yang memiliki missing values

In [5]:
df.isnull().sum()
df.isnull().mean()*100

,0
PassengerId,0.000000
Survived,0.000000
Pclass,0.000000
Name,0.000000
Sex,0.000000
Age,19.865320
SibSp,0.000000
Parch,0.000000
Ticket,0.000000
Fare,0.000000


Kolom yang memiliki missing values adalah Age, Cabin, dan Embarked.
Kolom Cabin memiliki jumlah missing paling banyak.
Persentase missing terbesar terdapat pada kolom Cabin, diikuti oleh Age dan Embarked.

Q3: Persentase penumpang yang selamat

In [6]:
df['Survived'].mean()*100

np.float64(38.38383838383838)

Persentase penumpang yang selamat adalah 38,3%

Q4: Penumpang wanita kelas 1

In [7]:
wk1 = df[(df['Sex']=='female') & (df['Pclass']==1)]

jumlah = wk1.shape[0]
rata_usia = wk1['Age'].mean()

jumlah, rata_usia

(94, np.float64(34.61176470588235))

Jumlah penumpang wanita kelas 1 adalah 94 orang.
Rata-rata usia penumpang wanita kelas 1 adalah sekitar 34 tahun.
Hal ini menunjukkan bahwa mayoritas penumpang wanita kelas 1 berada pada usia dewasa.

Q5: Tingkat keselamatan per kelas penumpang

In [8]:
df.groupby('Pclass')['Survived'].mean()

,Survived
Pclass,
1,0.629630
2,0.472826
3,0.242363


Tingkat keselamatan tertinggi terdapat pada penumpang kelas 1, diikuti kelas 2, dan paling rendah kelas 3.
Hal ini menunjukkan bahwa penumpang kelas atas memiliki peluang lebih besar untuk selamat dibandingkan kelas bawah.

Q6: Analisis tambahan berdasarkan jenis kelamin

In [9]:
df.groupby('Sex')['Survived'].mean()

,Survived
Sex,
female,0.742038
male,0.188908


Penumpang perempuan memiliki tingkat keselamatan yang jauh lebih tinggi dibandingkan laki-laki.
Hal ini menunjukkan adanya prioritas penyelamatan terhadap perempuan saat kejadian Titanic.